
  This patch corrects a Verify InitCatalog problem where in certain cases
  a garbage filename may be entered in the verification database.  This 
  fixes bug #1143.

  Apply the patch to version 2.4.2 (and previous versions) with:

  cd <bacula-source>
  patch -p0 <2.4.2-verify.patch
  ./configure <your-options>
  make
  ...
  make install


Index: src/dird/fd_cmds.c
===================================================================
--- src/dird/fd_cmds.c	(revision 7469)
+++ src/dird/fd_cmds.c	(working copy)
@@ -616,7 +616,6 @@
       char *p, *fn;
       char Digest[MAXSTRING];      /* either Verify opts or MD5/SHA1 digest */
 
-      jcr->fname = check_pool_memory_size(jcr->fname, fd->msglen);
       if ((len = sscanf(fd->msg, "%ld %d %s", &file_index, &stream, Digest)) != 3) {
          Jmsg(jcr, M_FATAL, 0, _("<filed: bad attributes, expected 3 fields got %d\n"
 "msglen=%d msg=%s\n"), len, fd->msglen, fd->msg);
@@ -641,7 +640,7 @@
             }
          }
          /* Any cached attr is flushed so we can reuse jcr->attr and jcr->ar */
-         fn = jcr->fname;
+         fn = jcr->fname = check_pool_memory_size(jcr->fname, fd->msglen);
          while (*p != 0) {
             *fn++ = *p++;                /* copy filename */
          }
